### PR TITLE
svt-av1: init at 0.8.6 // ffmpeg-full: enable svt-av1

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Encode with svt-av1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/ffmpeg-full/default.nix [hunk 1]
```diff
@@ -251,6 +252,12 @@ stdenv.mkDerivation rec {
       sha256 = "sha256-dqpmpDFETTuWHWolMoLaubU4BeDEuQaBNA0wmzL1f8o=";
       name = "fix_libsrt.patch";
     })
+    # Patch ffmpeg for svt-av1 until version 4.4
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/AOMediaCodec/SVT-AV1/v0.8.4/ffmpeg_plugin/0001-Add-ability-for-ffmpeg-to-run-svt-av1.patch";
+      sha256 = "1p4g8skr5gjw5h1648j7qrks81zx49lrnx9g0p81qgnrvxc2wwx0";
+      name = "enable_svt-av1.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ffmpeg-full/default.nix:259]:
```suggestion
```
I think the upstream name is fine.

**Review by SuperSandro2000 — COMMENTED** [review]
I would prefer to wait until the next release before adding this. They might find more bugs and fix them.

**Comment by Chiiruno** [comment]
cc @codyopel

File: pkgs/tools/video/svt-av1/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "AOMediaCodec";
+    repo = "SVT-AV1";
+    rev = "v${version}";
+    sha256 = "1gfxdmdql090p7c8ln1z344g467l41p45287lmwy590hv8gw6bfg";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ yasm ];
```
Comment (codyopel) [inline pkgs/tools/video/svt-av1/default.nix:15]: `yasm` should be in `nativeBuildInputs`, Their build system does support `nasm` and I would strongly recommend it over yasm.

Comment (Chiiruno) [reply pkgs/tools/video/svt-av1/default.nix]: Done.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -12642,6 +12644,7 @@ julia_15 = callPackage ../development/compilers/julia/1.5.nix {
     libmfx = if stdenv.isDarwin then null else intel-media-sdk;
     libpulseaudio = if stdenv.isDarwin then null else libpulseaudio;
     rav1e = if stdenv.isDarwin then null else rav1e;
+    svt-av1 = if stdenv.isDarwin then null else svt-av1;
```
Comment (marsam) [inline pkgs/top-level/all-packages.nix:12647]: This line is not necessary, it builds on darwin. Would you mind removing it?
Besides that, LGTM

**Comment by marsam** [comment]
@GrahamcOfBorg build svt-av1

**Comment by marsam** [comment]
@GrahamcOfBorg build svt-av1
@GrahamcOfBorg build ffmpeg-full

**Comment by marsam** [comment]
@GrahamcOfBorg build svt-av1
@GrahamcOfBorg build ffmpeg-full

**Pull request merged**